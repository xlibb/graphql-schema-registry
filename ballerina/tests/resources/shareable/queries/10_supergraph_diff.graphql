query Diff {
    diff(newVersion: "1.0.0", oldVersion: "0.1.0") {
        severity
        action
        subject
        location
        value
        fromValue
        toValue
    }
}
