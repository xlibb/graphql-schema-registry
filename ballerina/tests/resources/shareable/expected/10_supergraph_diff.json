{
    "data": {
        "diff": [
            {
                "severity": "SAFE",
                "action": "ADDED",
                "subject": "TYPE",
                "location": [],
                "value": "Bux",
                "fromValue": null,
                "toValue": null
            },
            {
                "severity": "SAFE",
                "action": "ADDED",
                "subject": "TYPE",
                "location": [],
                "value": "Waldo",
                "fromValue": null,
                "toValue": null
            },
            {
                "severity": "SAFE",
                "action": "ADDED",
                "subject": "FIELD",
                "location": ["Query"],
                "value": "value",
                "fromValue": null,
                "toValue": null
            },
            {
                "severity": "SAFE",
                "action": "ADDED",
                "subject": "FIELD_DESCRIPTION",
                "location": ["Query", "name"],
                "value": "This is the name description in subgraph-2",
                "fromValue": null,
                "toValue": null
            },
            {
                "severity": "BREAKING",
                "action": "CHANGED",
                "subject": "FIELD_TYPE",
                "location": ["Query", "age"],
                "value": null,
                "fromValue": "Int!",
                "toValue": "Int"
            },
            {
                "severity": "DANGEROUS",
                "action": "ADDED",
                "subject": "UNION_MEMBER",
                "location": ["UnionFoo"],
                "value": "Waldo",
                "fromValue": null,
                "toValue": null
            },
            {
                "severity": "DANGEROUS",
                "action": "ADDED",
                "subject": "UNION_MEMBER",
                "location": ["UnionFoo"],
                "value": "Bux",
                "fromValue": null,
                "toValue": null
            }
        ]
    }
}
